ste/xfer-anyToRemote-file.go

```diff
@@ -168,7 +168,7 @@ func anyToRemote(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, sen
 			dstRQ := dstURL.Query()
 
 			// note that query is now all lower case at this point
-			if len(srcRQ["snapshot"]) == 0 && len(srcRQ["versionid"]) == 0 && len(dstRQ["snapshot"]) == 0 && len(dstRQ["versionid"]) == 0 {
+			if len(srcRQ["sharesnapshot"]) == 0 && len(srcRQ["versionid"]) == 0 && len(dstRQ["sharesnapshot"]) == 0 && len(dstRQ["versionid"]) == 0 {
```

Comment:
Something feels fishy here. As far as I know, users can't write to old snapshots and versions. But, out of caution (in case the service were ever to allow such functionality), I feel like we should be comparing these items, INSTEAD of just checking they're nonzero.

Reply:
What do other contributors think?

Reply:
Hey @siminsavani-msft ,
I think @adreed-msft is right. This if-else condition is never going to execute because -
1. Snapshots are immutable copies taken at a point of time. We can read copy from one snapshot but not change them.
2. Similarly we cannot edit a particular version. Think of versions as a Singly LinkedList with head pointer pointing to the latest version. The only way we can update a version is to move the header back to that version except in this case, we create new version by discarding changes.

Reply:
Snapshot being immutable is universally accepted, and we *should* assume same as well. Comments on line 165/166 explain the rest. If the source and destination are same container/filestore, and if we're not reading from snapshot(or versionid) - throw that source and destination are same. Else, we'll let the service dictate error.

Reply:
This makes complete sense! The sharesnapshot & versionID of both source and destination should be compared instead of the current behavior where their length is compared to 0. Will update this! Thanks y'all!